packages/retail-ui-extensions/src/extension-api/CartApi/CartApiContent.ts

```diff
@@ -0,0 +1,102 @@
+
+export type DiscountType = 'Percentage' | 'FixedAmount' | 'Code';
+
+export interface CartApiContent {
+  /** Initial value of the Cart when the screen is rendered. */
+  initialValue: Cart;
+
+  /** Callback when the cart changes */
+  setOnChange(onChange: (cart: Cart) => void): void;
+
+  /** Apply a cart level discount
+   * @param type the type of discount applied (example: 'Percentage')
+   * @param title the title attributed with the discount
+   * @param amount the percentage or fixed monetary amount deducted with the discout
+   */
+  applyCartDiscount(type: DiscountType, title: string, amount: string): void;
+
+  /** Remove the cart discount */
+  removeCartDiscount(): void;
+
+  /** Clear the cart */
+  clearCart(): void;
+
+  /** Set the customer in the cart
+   * @param customer the customer object to add to the cart
+   */
+  setCustomer(customer: Customer): void;
+
+  /** Remove the current customer from the cart */
+  removeCustomer(): void;
+
+  /** Add a custom sale to the cart
+   * @param customSale the custom sale object to add to the cart
+   */
+  addCustomSale(customSale: CustomSale): void;
+
+  /** Add a line item by variant ID to the cart
+   * @param variantId the product variant's numeric ID to add to the cart
+   * @param quantity the number of this variant to add to the cart
+   */
+  addLineItem(variantId: number, quantity: number): void;
+
+  /** Remove the line item at this index from the cart
+   * @param index the index of the line item that should be removed
+   */
+  removeLineItem(index: number): void;
+
+  /** Adds custom properties to the cart
+   * @param properties the custom key to value object to attribute to the cart
+   */
+  addCartProperties(properties: Record<string, string>): void;
+
+  /** Removes the specified cart properties
+   * @param keys the collection of keys to be removed from the cart properties
+   */
+  removeCartProperties(keys: string[]): void;
+
+  /** Adds custom properties to the specified line item
+   * @param index the index of the line item to which the properties should be added
+   * @param properties the custom key to value object to attribute to the line item
+   */
+  addLineItemProperties(
+    index: number,
+    properties: Record<string, string>,
+  ): void;
+
+  /** Removes the specified line item properties
+   * @param index the index of the line item to which the properties should be removed
+   * @param keys the collection of keys to be removed from the line item properties
+   */
+  removeLineItemProperties(index: number, keys: string[]): void;
+
+  /** Add a discount on a line item to the cart
+   * @param index the index of the line item that should receive a discount
+   * @param type the type of discount applied (example: 'Percentage')
+   * @param title the title attributed with the discount
+   * @param amount the percentage or fixed monetary amount deducted with the discout
+   */
+  setLineItemDiscount(
+    index: number,
```

Comment:
The final API is up to you folks, but I would encourage you to avoid the very positional argument-heavy structure you have for these APIs. They are very fragile as adding properties after the fact is very difficult. Most other APIs we design (for UI extensions and just in general) rely much more on object arguments, where adding properties is easy.

Reply:
Good feedback. So far most of these functions are based on what partners currently use for App Bridge, but we are definitely going to be trying to make that experience better at the same time.